src/java/org/apache/cassandra/index/sai/disk/ScoreCacheMapper.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import com.carrotsearch.hppc.LongFloatHashMap;
+import org.apache.cassandra.index.sai.QueryContext;
+import org.apache.cassandra.index.sai.utils.PrimaryKey;
+
+@NotThreadSafe
+public class ScoreCacheMapper
```

Comment:
This is much better. If I was going to have one nit, it is that we could follow the pattern we have elsewhere and make `ScoreCacheMapper` an interface with a `static create` method and have an `EMPTY` and `Impl` implementations. This would avoid the `if (cache != null)` for every map call.

Feel free to ignore this.

Reply:
done